monai/engines/workflow.py

```diff
@@ -125,8 +124,6 @@ def __init__(
             super().__init__(iteration_update)
         else:
             super().__init__(self._iteration)
-        if not isinstance(device, torch.device):
-            raise TypeError(f"Device must be a torch.device but is {type(device).__name__}.")
```

Comment:
If removing this check, should we also change the `typehint` and use `engine.state.device = torch.device(device)`?

Thanks.